src/Diff/index.js

```diff
@@ -23,112 +23,111 @@ const setUserSelectStyle = (element, selectable) => {
+    );
+
+    const {diffType, children, className, optimizeSelection, hunks, ...remainings} = props;
+    const {gutterType, viewType} = remainings;
+    const hideGutter = gutterType === 'none';
+    const monotonous = diffType === 'add' || diffType === 'delete';
+    const onTableMouseDown = (viewType === 'split' && !monotonous && optimizeSelection) ? enableColumnSelection : noop;
+    const cols = ((viewType, monotonous) => {
+        if (viewType === 'unified') {
             return (
                 <colgroup>
+                    {!hideGutter && <col className="diff-gutter-col" />}
                     {!hideGutter && <col className="diff-gutter-col" />}
                     <col />
+                </colgroup>
+            );
+        }
+
+        if (monotonous) {
+            return (
+                <colgroup>
                     {!hideGutter && <col className="diff-gutter-col" />}
                     <col />
                 </colgroup>
             );
-        })(props.viewType, monotonous);
+        }
 
         return (
-            <Provider value={{...props, monotonous}}>
-                <table
-                    ref={element => (this.root = element)}
-                    className={classNames('diff', `diff-${viewType}`, className)}
-                    onMouseDown={onTableMouseDown}
-                >
-                    {cols}
-                    {children(hunks)}
-                </table>
-            </Provider>
+            <colgroup>
+                {!hideGutter && <col className="diff-gutter-col" />}
+                <col />
+                {!hideGutter && <col className="diff-gutter-col" />}
+                <col />
+            </colgroup>
         );
-    }
-}
+    })(props.viewType, monotonous);
+
+    return (
+        <Provider value={{...remainings, monotonous}}>
+            <table
+                ref={root}
+                className={classNames('diff', `diff-${viewType}`, className)}
+                onMouseDown={onTableMouseDown}
+            >
+                {cols}
+                {children(hunks)}
+            </table>
+        </Provider>
+    );
+};
+
+Diff.propTypes = {
+    diffType: PropTypes.oneOf(['add', 'delete', 'modify', 'rename', 'copy']).isRequired,
+    viewType: PropTypes.oneOf(['unified', 'split']).isRequired,
+    hunks: PropTypes.arrayOf(PropTypes.object).isRequired,
+    gutterType: PropTypes.oneOf(['default', 'none', 'anchor']),
+    generateAnchorID: PropTypes.func,
+    selectedChanges: PropTypes.arrayOf(PropTypes.string),
+    widgets: PropTypes.objectOf(PropTypes.node),
+    optimizeSelection: PropTypes.bool,
+    className: PropTypes.string,
+    renderToken: PropTypes.func,
+    children: PropTypes.func,
+};
+
+Diff.defaultProps = {
+    gutterType: 'default',
+    optimizeSelection: false,
+    selectedChanges: [],
+    widgets: {},
+    className: '',
+    renderToken: undefined,
+    generateAnchorID() {
+        return undefined;
+    },
+    children(hunks) {
```

Comment:
It is wired to have a prop named children, and it is a function.

Migrate it later?

Reply:
This may not be changeable, it could cause a incompatible change, it's likely a render prop named `children`, like react context's `<Consumer>`